### ISSUE TITLE
Panic when using forall in `ensures` of function containing `return`

### ISSUE THREAD

**Issue by matthias-brun** [verus-lang/verus#216]
This might be related to #215. The following code causes a panic. If `return true;` is replaced with `true` or if we use an ensures without a quantifier, it verifies without issues.

```rust
#[allow(unused_imports)]
use builtin::*;
mod pervasive;

#[spec]
fn f(a: nat) -> bool {
    true
}

fn g() -> bool {
    ensures(|res: bool| forall(|i: nat| f(i)));
    return true;
}

fn main() {
}
```
results in
```
fhs m: source 1 ν ./tools/rust-verify.sh /st/verus/test.rs
note: verifying root module

thread 'rustc' panicked at 'assertion failed: trigs.len() == 0', vir/src/ast_to_sst.rs:227:21
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
verification results:: verified: 0 errors: 0
```

**Comment by tjhance** [comment]
Currently, expression-return and return statements are handled in different places, the first in func_to_air and the second in ast_to_sst. The result is there are a few differences in the way they're handled.

At one point I attempted to consolidate them into place to resolve the inconsistencies,  but I never followed through as it turned out to be kind of tricky and there were a few things i didn't know how to handle. It's probably time to revisit it, though.

**Comment by tjhance** [comment]
fixed by #312
